baremetal/metal3machine_manager.go

```diff
@@ -416,60 +413,12 @@ func (m *MachineManager) getUserData(ctx context.Context, host *bmh.BareMetalHos
 			Namespace: m.Machine.Namespace,
 		}
 		return nil
-
-	} else if m.Machine.Spec.Bootstrap.Data == nil {
-		// If we do not have DataSecretName or Data then exit
-		return nil
-
-	}
-
-	// If we have Data, use it
-	decodedUserData := *m.Machine.Spec.Bootstrap.Data
-	// decode the base64 cloud-config
-	decodedUserDataBytes, err = base64.StdEncoding.DecodeString(decodedUserData)
-	if err != nil {
-		return err
-	}
-
-	err = m.createSecret(ctx, m.Metal3Machine.Name+"-user-data",
-		m.Metal3Machine.Namespace, map[string][]byte{
-			"userData": decodedUserDataBytes,
-		},
-	)
-	if err != nil {
-		return err
 	}
 
 	m.Metal3Machine.Status.UserData = &corev1.SecretReference{
```

Comment:
why do you still have this here? It appears to point to nothing..

Reply:
Fixed, thank you.